test/gelatinous_cube/api_test.clj

```diff
@@ -97,7 +97,10 @@
                   {:succeeded-norms all-norm-names}))
   (with-redefs [impl/transact-norm! (fn [& _] (throw (ex-info "!" {})))]
     (is (thrown-with-data?
-         {:unneeded-norms immutable-norm-names
+         {:unneeded-norms (->> config
+                               (partition-by :mutable)
+                               (first)
+                               (map :name))
```

Comment:
I think this is more brittle than relying on the `immutable-norm-names` var as it assumes a particular sequence of mutable/immutable norm maps that doesn't have to be true. The use of `unneeded-norms` here was just so we could assert against a more sophisticated return shape than simulating an exception on the first pass for this db.

Reply:
I made this change because the value of `:unneeded-norms` is dependent on the ordering of norm absorption, because of [this `throw` during `reduce`](https://github.com/daemianmack/gelatinous-cube/blob/b8733829776f161c0ed4dcfe1415444495223e38/src/gelatinous_cube/api.clj#L27). Without this change adding an immutable norm after the first mutable breaks the test, but adding it before does not.

Did you intend for absorption to short-circuit in this case or should it attempt all norms?